src/components/Icon/index.js

```diff
@@ -1,29 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 
 function Icon({
   className,
   family,
   type,
+  hoverIconType,
   ...rest
 }) {
-  function getCSSClasses() {
+  const [isHovered, setIsHovered] = useState(false);
+  function getCSSClasses(showHoverIcon) {
     const fontFamily = family || 'fa';
 
     return [
       'tyk-icon',
       fontFamily,
-      `${fontFamily}-${type}`,
+      `${fontFamily}-${showHoverIcon ? hoverIconType : type}`,
       className,
     ].filter(Boolean).join(' ');
   }
 
-  return <i className={getCSSClasses()} {...rest} />;
+  const toggleHover = () => setIsHovered(!isHovered);
```

Comment:
I would use `useCallback` for this function and also would use the function form for changing the `isHovered` state.
```suggestion
  const toggleHover = useCallback(() => setIsHovered(h => !h), []);
```